src/particle.f90

```diff
@@ -0,0 +1,2351 @@
+
+
+    if(abs(plx)<1.d-16 .and. abs(ply)<1.d-16 .and. abs(plz)<1.d-16) then
+
+        npx=1; npy=1; npz=1
+
+    elseif(abs(plx)<1.d-16 .and. abs(ply)<1.d-16) then
+
+        npx=1
+        npy=1
+        npz=nptotal
+
+    elseif(abs(plx)<1.d-16 .and. abs(plz)<1.d-16) then
+
+        npx=1
+        npy=nptotal
+        npz=1
+
+    elseif(abs(ply)<1.d-16 .and. abs(plz)<1.d-16) then
+
+        npx=nptotal
+        npy=1
+        npz=1
+
+    elseif(abs(plx)<1.d-16) then
+
+        n1=sqrt(real(nptotal,mytype)/(ply*plz))
+        
+        npx=1 
+        npy=int(n1*ply)
+        npz=int(n1*plz)
+
+    elseif(abs(ply)<1.d-16) then
+
+        n1=sqrt(real(nptotal,mytype)/(plx*plz))
+        
+        npx=int(n1*plx)
+        npy=1
+        npz=int(n1*plz)
+
+    elseif(abs(plz)<1.d-16) then
+
+        n1=sqrt(real(nptotal,mytype)/(plx*ply))
+        
+        npx=int(n1*plx)
+        npy=int(n1*ply)
+        npz=1
+
+    else
+
+        n1=(real(nptotal,mytype)/(plx*ply*plz))**(1.0/3.0)
+        npx=int(n1*plx)
+        npy=int(n1*ply)
+        npz=int(n1*plz)
+
+    endif
+
+  end subroutine particle_dis_xyz
+  !+-------------------------------------------------------------------+
+  !| The end of the subroutine particle_dis_xyz.                       |
+  !+-------------------------------------------------------------------+
+
+  !+-------------------------------------------------------------------+
+  !| This subroutine is to calcualte the size and range of local domain|
+  !+-------------------------------------------------------------------+
+  !| CHANGE RECORD                                                     |
+  !| -------------                                                     |
+  !| 15-11-2021  | Created by J. Fang                                  |
+  !+-------------------------------------------------------------------+
+  subroutine local_domain_size
+    !
+    use param,     only : dx,dy,dz,istret
+    use variables, only : yp,ny
+    use decomp_2d, only : xstart,xend
+    use actuator_line_model_utils
+    !
+    integer :: nyr,nzr,jrank
+    !
+    allocate( lxmin(0:nproc-1),lxmax(0:nproc-1),  &
+              lymin(0:nproc-1),lymax(0:nproc-1),  &
+              lzmin(0:nproc-1),lzmax(0:nproc-1)   )
+    !
+    lxmin=0._mytype; lxmax=0._mytype
+    lymin=0._mytype; lymax=0._mytype
+    lzmin=0._mytype; lzmax=0._mytype
+    !
+    lxmin(nrank)=real(0,mytype)*dx
+    lxmax(nrank)=real(xend(1),mytype)*dx
+    !
+    if (istret==0) then
+      lymin(nrank)=real(xstart(2)-1,mytype)*dy
+      lymax(nrank)=real(xend(2),mytype)*dy
+    else
+      lymin(nrank)=yp(xstart(2))
+      nyr=min(ny,xend(2)+1)
+      lymax(nrank)=yp(nyr)
+    endif
+    !
+    lzmin(nrank)=real((xstart(3)-1),mytype)*dz
+    nzr=xend(3)
+    lzmax(nrank)=real(nzr,mytype)*dz
+    !
+    lxmin=psum(lxmin); lxmax=psum(lxmax)
+    lymin=psum(lymin); lymax=psum(lymax)
+    lzmin=psum(lzmin); lzmax=psum(lzmax)
+    !
+     ! do jrank=0,nproc-1
```

Comment:
Please do not commit debug code using comments. I would suggest to remove completely debug code. If you really want to include debug code in your PR :

```
#ifdef DEBUG
if (.false.) then
```
Your debug code
```
endif
#endif
```